### PR TITLE
fix: returns 0 exit code on error

### LINKED ISSUE #1315: Gitui returns zero exit code on error
**Describe the bug**

When calling `gitui` in a non-git directory, I get the expected error:
```shell
> gitui
invalid path
please run gitui inside of a non-bare git repository
```
However the exit code is zero:
```shell
> echo $?
0
```

**Expected behavior**
I'd expect a non-zero exit code e.g. `1`.

**Context (please complete the following information):**
This is on macOs 12.5.1 using `gitui` installed via `nix`
```
> gitui --version
gitui 0.21.0
```

**Additional context**
This is a problem when scripting shortcuts to call `gitui` in a directory as errors get swallowed.

### PR DESCRIPTION
**Pull request by snapdgn**
This Pull Request fixes/closes #1315.

It changes the following:
- changes exit code to 1, while initializing gitui in a bare git repo (let me know if the exit code needs to be something else)

I followed the checklist:
- [ ] I added unittests
- [x] I ran `make check` without errors
- [x] I tested the overall application
- [ ] I added an appropriate item to the changelog

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -113,7 +113,7 @@ fn main() -> Result<()> {
 
 	if !valid_path(&cliargs.repo_path) {
 		eprintln!("invalid path\nplease run gitui inside of a non-bare git repository");
-		return Ok(());
+		process::exit(1);
```
Comment (extrawurst) [inline src/main.rs:116]: Did you check if there other places that should return non zero?

Comment (snapdgn) [reply src/main.rs:116]: If you mean, checking at all potential places for all kinds of errors which should've returned a non-zero exit code. Sorry, I haven't. Will do :)

Comment (SimonAdameit) [reply src/main.rs:116]: Thanks for tackling this! Wouldn't it be more ideomatic to use `return Err(..)` or `bail!(...)` here, as `main` returns a `Result`? I think an error should get turned into exit code `1` automatically.

**Review by SimonAdameit — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (snapdgn) [reply src/main.rs:116]: hey, thanks a lot for the review. I completely agree with you, will update this accordingly. Also, please do let me know, if there are any other places i could potentially look into, would be very grateful. Thanks!

**Comment by extrawurst** [comment]
Thanks!

**Pull request merged**